implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -53,11 +53,22 @@ public JwtContext parse(final String token, final JWTAuthContextInfo authContext
         try {
             JwtConsumerBuilder builder = new JwtConsumerBuilder()
                     .setRequireExpirationTime()
-                    .setRequireSubject()
-                    .setSkipDefaultAudienceValidation()
-                    .setJwsAlgorithmConstraints(
-                            new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
-                                    AlgorithmIdentifiers.RSA_USING_SHA256));
+                    .setSkipDefaultAudienceValidation();
+
+            if (authContextInfo.getDefaultSubjectClaim() == null ||
+                    Claims.sub.name().equals(authContextInfo.getDefaultSubjectClaim())) {
+                builder.setRequireSubject();
+            }
+
+            if (authContextInfo.getWhitelistAlgorithms().isEmpty()) {
+                builder.setJwsAlgorithmConstraints(
+                        new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
+                                AlgorithmIdentifiers.RSA_USING_SHA256));
+            } else {
+                builder.setJwsAlgorithmConstraints(
```

Comment:
I'd like to suggest to do more work around it:

- please use `AlgorithmIdentifiers.valueOf(String).name()` or similar to ensure these are valid values
- please filter out the symmetric algorithms [starting from HS](https://bitbucket.org/b_c/jose4j/src/d05e68f9822cd479f2d42edfa5436d3d749ae3aa/src/main/java/org/jose4j/jws/AlgorithmIdentifiers.java#lines-25). As I've mentioned earlier MP JWT does not like the fact the secret key has to be visible to both the client and all the microservices nodes which does not scale and is problematic from the key management POV.

Reply:
Yes, I was still not done with this. 

I was planning to add a setWhitelistAlgorithms method to SmallryeJwtUtils and perform validation regarding the accepted algorithms types and output a log message if you are setting an unsupported algorithm.